File: backend/src/main/java/com/eventeasy/service/AuthService.java
package com.eventeasy.service;

import com.eventeasy.dto.request.ChangePasswordRequest;
import com.eventeasy.dto.request.LoginRequest;
import com.eventeasy.dto.request.UpdateProfileRequest;
import com.eventeasy.dto.response.AuthResponse;
import com.eventeasy.dto.response.UserResponse;

/**
 * <p>Service interface declaring authentication workflow operations for EventEasy application.</p>
 * <p>Contains contract signatures for user login, current user retrieval, profile updating, and password changing.</p>
 *
 * @author EventEasy Engineering
 * @version 1.0.0
 */
public interface AuthService {

    /**
     * Authenticate user credentials and return signed JWT access token with profile payload.
     *
     * @param loginRequest login credentials request payload
     * @return AuthResponse containing access token and UserResponse
     */
    AuthResponse login(LoginRequest loginRequest);

    /**
     * Retrieve current authenticated user profile details from SecurityContext.
     *
     * @return UserResponse profile payload
     */
    UserResponse getCurrentUser();

    /**
     * Update current authenticated user's email, first name, and last name.
     *
     * @param request profile update details
     * @return updated UserResponse profile
     */
    UserResponse updateProfile(UpdateProfileRequest request);

    /**
     * Change current authenticated user's password verifying current password match.
     *
     * @param request password change details
     */
    void changePassword(ChangePasswordRequest request);
}

File: frontend/src/services/authService.js
import axiosClient from '../api/axiosClient';
import { API_ENDPOINTS } from '../constants/apiEndpoints';

/**
 * Authentication & Admin Settings API Service wrapping backend HTTP calls.
 */
export const authService = {
  /**
   * Authenticate admin credentials and return token and profile.
   * @param {Object} credentials { email, password }
   * @returns {Promise<Object>} AuthResponse data payload ({ accessToken, user })
   */
  login: async (credentials) => {
    const response = await axiosClient.post(API_ENDPOINTS.AUTH.LOGIN, credentials);
    return response?.data ? response.data : response;
  },

  /**
   * Fetch current authenticated admin user profile.
   * @returns {Promise<Object>} UserResponse profile payload
   */
  getCurrentUser: async () => {
    const response = await axiosClient.get(API_ENDPOINTS.AUTH.ME);
    return response?.data ? response.data : response;
  },

  /**
   * Update administrator profile details (email, first name, last name).
   * @param {Object} profileData { email, firstName, lastName }
   * @returns {Promise<Object>} Updated UserResponse profile
   */
  updateProfile: async (profileData) => {
    const response = await axiosClient.put('/auth/profile', profileData);
    return response?.data ? response.data : response;
  },

  /**
   * Change administrator account password.
   * @param {Object} passwordData { currentPassword, newPassword }
   * @returns {Promise<Object>} API response
   */
  changePassword: async (passwordData) => {
    const response = await axiosClient.put('/auth/password', passwordData);
    return response?.data ? response.data : response;
  },
};
